Broken Units with recent changes (java.lang.Exception: megamek.common.loaders.EntityLoadingException:)

I suspect it might be related to a recent PR.

### Environment
What version of MegaMek does your issue apply to?  0.49.19 from the Repo as off todays date.
What operating system are you using?  Windows and Java 17

### Description
<img width="581" alt="image" src="https://github.com/MegaMek/megamek/assets/13752452/6cfbd633-4a42-4c86-a05c-c0ca643c4856">

### Files
The units are correct and will no longer load.

```
10:29:44,111 ERROR [megamek.common.MechFileParser] {Thread-0}
megamek.common.MechFileParser.<init>(MechFileParser.java:48) - 
java.lang.Exception: megamek.common.loaders.EntityLoadingException: Improved C3 Computer (C3I) does not fit in CT on Cronus CNS-TD9 ID:-1
        free criticals in location: 1, criticals needed: 2
    at megamek.common.loaders.MtfFile.getEntity(MtfFile.java:495)
    at megamek.common.MechFileParser.parse(MechFileParser.java:157)
    at megamek.common.MechFileParser.<init>(MechFileParser.java:46)
    at mekhq.campaign.unit.UnitTechProgression.calcTechProgression(UnitTechProgression.java:122)
    at mekhq.campaign.unit.UnitTechProgression$BuildMapTask.call(UnitTechProgression.java:150)
    at mekhq.campaign.unit.UnitTechProgression$BuildMapTask.call(UnitTechProgression.java:138)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: megamek.common.loaders.EntityLoadingException: Improved C3 Computer (C3I) does not fit in CT on Cronus CNS-TD9 ID:-1
        free criticals in location: 1, criticals needed: 2
    at megamek.common.loaders.MtfFile.parseCrits(MtfFile.java:922)
    at megamek.common.loaders.MtfFile.getEntity(MtfFile.java:405)
    ... 7 more

10:29:44,377 ERROR [megamek.common.loaders.MtfFile] {Thread-0}
megamek.common.loaders.MtfFile.getEntity(MtfFile.java:494) - 
megamek.common.loaders.EntityLoadingException: MASC does not fit in LT on Tundra Wolf 2 ID:-1
        free criticals in location: 2, criticals needed: 3
    at megamek.common.loaders.MtfFile.parseCrits(MtfFile.java:922)
    at megamek.common.loaders.MtfFile.getEntity(MtfFile.java:405)
    at megamek.common.MechFileParser.parse(MechFileParser.java:157)
    at megamek.common.MechFileParser.<init>(MechFileParser.java:46)
    at mekhq.campaign.unit.UnitTechProgression.calcTechProgression(UnitTechProgression.java:122)
    at mekhq.campaign.unit.UnitTechProgression$BuildMapTask.call(UnitTechProgression.java:150)
    at mekhq.campaign.unit.UnitTechProgression$BuildMapTask.call(UnitTechProgression.java:138)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/megamek/src/megamek/common/Mech.java
+++ b/megamek/src/megamek/common/Mech.java
@@ -2745,6 +2745,18 @@ public abstract class Mech extends Entity {
         }
     }
 
+    /**
+     * Adds equipment without adding slots for it.
+     * Specifically for targeting computers, which when loaded from a file don't have a correct size and get loaded slot by slot
+     */
+    public Mounted addTargCompWithoutSlots(EquipmentType etype, int loc, boolean omniPod, boolean armored) throws LocationFullException {
+        Mounted mounted = new Mounted(this, etype);
+        mounted.setOmniPodMounted(omniPod);
+        mounted.setArmored(armored);
+        super.addEquipment(mounted, loc, false);
+        return mounted;
+    }
+
     public Mounted addEquipment(EquipmentType etype, EquipmentType etype2,
             int loc,  boolean omniPod, boolean armored) throws LocationFullException {
         Mounted mounted = new Mounted(this, etype);

--- a/megamek/src/megamek/common/loaders/MtfFile.java
+++ b/megamek/src/megamek/common/loaders/MtfFile.java
@@ -800,15 +800,15 @@ public class MtfFile implements IMechLoader {
                                               isTurreted);
                         m.setOmniPodMounted(isOmniPod);
                         hSharedEquip.put(etype, m);
-                    } else if (etype.hasFlag(MiscType.F_TARGCOMP)) {
+                    } else if (etype instanceof  MiscType && etype.hasFlag(MiscType.F_TARGCOMP)) {
                         // Targeting computers are special, they need to be loaded like spreadable equipment, but they aren't spreadable
                         Mounted m = hSharedEquip.get(etype);
                         if (m == null) {
-                            m = mech.addEquipment(etype, loc, rearMounted, BattleArmor.MOUNT_LOC_NONE, isArmored, isTurreted);
-                            m.setOmniPodMounted(isOmniPod);
+                            m = mech.addTargCompWithoutSlots(etype, loc, isOmniPod, isArmored);
                             hSharedEquip.put(etype, m);
                         }
                         mech.addCritical(loc, new CriticalSlot(m));
+
 
                     } else if (((etype instanceof WeaponType) && ((WeaponType) etype).isSplitable()) || ((etype instanceof MiscType) && etype.hasFlag(MiscType.F_SPLITABLE))) {
                         // do we already have this one in this or an outer location?

